lib/tests/streamlit/commands/navigation_test.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import streamlit as st
+from streamlit.errors import StreamlitAPIException
+from tests.delta_generator_test_case import DeltaGeneratorTestCase
+
+
+class NavigationTest(DeltaGeneratorTestCase):
+    """Test st.navigation"""
+
+    def test_no_pages(self):
+        """Test that an error is thrown with no pages"""
+        with self.assertRaises(StreamlitAPIException):
+            st.navigation([])
+
+    def test_single_page(self):
+        """Test that a single page is returned"""
+        single_page = st.Page("page1.py")
+        page = st.navigation([single_page])
+        self.assertEqual(page, single_page)
```

Comment:
nit: we've been slowly trying to move to using `assert x == y` instead of `self.assertEqual(x, y)` (and similar for all `self.assert*` methods) throughout the codebase since pytest formats asserts nicely for us. We don't always remember to change things when adding asserts to a test file that primarily uses `self.assert*`, but any newly added file should definitely changed to use raw asserts

Same comment to all other new test files in the PR.

Reply:
Ah. Good to note. Side question, are we keeping track of these initiatives. Would be good to setup.